tools/dotnet-linker/SetupStep.cs

```diff
@@ -27,6 +27,17 @@ public class SetupStep : ConfigurationAwareStep {
 			}
 		}
 
+		List<IMarkHandler> _markHandlers;
+		List<IMarkHandler> MarkHandlers {
+			get {
+				if (_markHandlers == null) {
+					var pipeline = typeof (LinkContext).GetProperty ("Pipeline").GetGetMethod ().Invoke (Context, null);
+					_markHandlers = (List<IMarkHandler>) pipeline.GetType ().GetProperty ("MarkHandlers").GetValue (pipeline);
```

Comment:
is there any plan to make those API public ?

Reply:
@sbomer why is it needed?

Reply:
This was just to get something working - I am looking into passing all steps individually as --custom-step args.

Reply:
SetupStep currently logs the pipeline steps which is one case where making this API public could be useful:
https://github.com/xamarin/xamarin-macios/blob/993c7443476b78bbee2a0c7fe7dcebb73778850f/tools/dotnet-linker/SetupStep.cs#L116-L123

Reply:
I'd like to see that in linker together with timing details under verbose mode

Reply:
Per discussion with @spouliot and @rolfbjarne the plan is to remove the reflection once https://github.com/mono/linker/issues/1314 is fixed, since there's no good workaround for sharing state without it. I also filed https://github.com/mono/linker/issues/2013 about the suggestion to output diagnostic info.

Reply:
I had quick look mono/linker#1314 won't be probably fixed anytime soon as it needs to introduce complex ALC handling. We could just add generic shared state holder to LinkContext and use that instead.

Reply:
> We could just add generic shared state holder to LinkContext and use that instead.

That won't solve the problem where the custom step assembly has a different identity every time it's loaded (so you can't use custom types when storing state).